.gitlab-ci.yml

```diff
@@ -42,6 +42,9 @@ parthenon-cuda-short:
       -DKokkos_ENABLE_CUDA_LAMBDA=True
       -DCMAKE_CXX_COMPILER=${PWD}/../external/Kokkos/bin/nvcc_wrapper
       ../ && make -j${J} && make -j${J} test
+  artifacts:
+    paths:
+      - build-cuda/CMakeFiles/CMakeOutput.log
```

Comment:
If I think you want `when: always`, don't you?

Reply:
Or even `on_failure`, but I agree that the implicit default `on_success` here is not what satisfies "When debugging failed CI"

Reply:
Also given that this artifact is meant for debugging I also suggest to allow cleaning sooner than later by adding `expire_in: 3d`

Reply:
Good idea. I'll set it to `when: always` and `expire_in: 3d`. I'd prefer `when: always` over `when: on_failure`, because sometimes it's nice to look at things like this, even if the tests pass.